src/ArmTemplates/Creator/TemplateCreators/TagTemplateCreator.cs

```diff
@@ -24,51 +26,64 @@ public TagTemplateCreator(ITemplateBuilder templateBuilder)
 
         public Template CreateTagTemplate(CreatorParameters creatorConfig)
```

Comment:
Can we return `Template<TTemplateResource>` here? I am not sure it is very important right now, but it allows us to reuse already handled resources by future templateCreators. The same is done in extractor

What do you think?